scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -1029,12 +1029,65 @@ public void changeJobPriority(JobId jobId, JobPriority priority)
         schedulingService.changeJobPriority(jobId, priority);
     }
 
+    /**
+     * This method blocks until job is actually finished its submission.
+     * @param jobId
+     */
+    private void blockUntilJobIsUnderSubmission(JobId jobId) {
+        while (waitForJobsToBeSubmitted.contains(jobId)) {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                logger.warn("Something tried to interrupt our blocking", e);
+            }
+        }
+    }
+
+    /**
+     * This method is related to submission process.
+     * This method should be called as soon as we received job id.
+     * If job was already finished its submition before this method, than job state will not change.
+     * Otherwise, if job is not yet finished its submition, than job state will be UNDER SUBMISSION.
+     * @param jobId
+     */
+    private void registerJobAsUnderSumbission(JobId jobId) {
+        synchronized (waitForJobsToBeSubmitted) {
+            synchronized (alreadySumbittedJobs) {
+                if (alreadySumbittedJobs.contains(jobId)) {
+                    alreadySumbittedJobs.remove(jobId);
+                    waitForJobsToBeSubmitted.remove(jobId);
+                } else {
+                    waitForJobsToBeSubmitted.add(jobId);
+                }
+            }
+        }
+    }
+
+    /**
+     * This method is related to submission process.
+     * This method should be called as soon as we finished to submit a job.
+     * @param jobId
+     */
+    private void unregisterSumbittedJob(JobId jobId) {
+        synchronized (waitForJobsToBeSubmitted) {
+            synchronized (alreadySumbittedJobs) {
+                if (waitForJobsToBeSubmitted.contains(jobId)) {
+                    alreadySumbittedJobs.remove(jobId);
+                    waitForJobsToBeSubmitted.remove(jobId);
+                } else {
+                    alreadySumbittedJobs.add(jobId);
+                }
+            }
+        }
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
     @ImmediateService
     public JobState getJobState(JobId jobId) throws NotConnectedException, UnknownJobException, PermissionException {
+        blockUntilJobIsUnderSubmission(jobId);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)